homeassistant/components/switcher_kis/_service_registration.py

```diff
@@ -0,0 +1,371 @@
+                        device_password, COMMAND_ON, timer)
+            return None
+
+        async def async_set_auto_off_service(service: ServiceCall) -> None:
+            """Use for handling setting device auto-off service calls."""
+            if SWITCH_ENTITY:
+                await set_auto_off_to_device(
+                    SWITCH_ENTITY.device_ip_addr, phone_id, device_id,
+                    device_password, service.data[CONF_AUTO_OFF])
+
+            return None
+
+        async def async_update_name_service(service: ServiceCall) -> None:
+            """Use for handling update device name service calls."""
+            if SWITCH_ENTITY:
+                await update_name_of_device(
+                    SWITCH_ENTITY.device_ip_addr, phone_id, device_id,
+                    device_password, service.data[CONF_NAME])
+
+            return None
+
+        for service in [SERVICE_TURN_ON, SERVICE_TURN_OFF, SERVICE_TURN_ON_15,
+                        SERVICE_TURN_ON_30, SERVICE_TURN_ON_45,
+                        SERVICE_TURN_ON_60]:
+            hass.services.async_register(
+                DOMAIN, service, async_switcher_control_service, schema={})
+
+        hass.services.async_register(
+            DOMAIN, SERVICE_SET_AUTO_OFF,
+            async_set_auto_off_service,
+            schema=SET_AUTO_OFF_SERVICE_SCHEMA)
+
+        hass.services.async_register(
+            DOMAIN, SERVICE_UPDATE_DEVICE_NAME,
+            async_update_name_service,
+            schema=UPDATE_DEVICE_NAME_SERVICE_SCHEMA)
+
+    async def async_sensor_platform_discovered(platform: str,
+                                               discovery_info: Optional[Dict]
+                                               ) -> None:
+        """Use for registering services after switch platform discoverd."""
+        async def async_create_schedule_service(service: ServiceCall) -> None:
+            """Use for handling create schedule service calls."""
+            recurring = service.data.get(CONF_RECURRING)
+            if recurring and not service.data.get(CONF_DAYS):
+                _LOGGER.error("A recursive schedule must contain %s",
+                              "a list of days to run at")
+            else:
+                requested_days = [0]
+                if recurring:
+                    for day in service.data[CONF_DAYS]:
+                        requested_days.append(DAYS_INT_DICT[day])
+
+                weekdays = create_weekdays_value(requested_days)
+                start_time = timedelta_str_to_schedule_time(
+                    service.data[CONF_START_TIME])
+                end_time = timedelta_str_to_schedule_time(
+                    service.data[CONF_END_TIME])
+
+                schedule_data = SCHEDULE_CREATE_DATA_FORMAT.format(
+                    weekdays, start_time, end_time)
+
+                if SWITCH_ENTITY:
+                    response = await create_schedule(
```

Comment:
While defined as `async` the implementation of this method calls blocking synchronous methods. This will need to be fixed in the library.